plugins/single_plugins/expo.cpp

```diff
@@ -403,6 +440,153 @@ class wayfire_expo : public wf::plugin_interface_t
         update_target_workspace(to.x, to.y);
     }
 
+    /**
+     * Helper to determine if keyboard presses should be handled
+     */
+    bool should_handle_key()
+    {
+        return ((!zoom_animation.running() || state.zoom_in) &&
+            keyboard_interaction && !state.button_pressed);
+    }
+
+    void handle_key_pressed(uint32_t key)
+    {
+        int x = target_vx;
+        int y = target_vy;
+
+        switch (key)
+        {
+          case KEY_ENTER:
+            deactivate();
+            return;
+
+          case KEY_ESC:
+            target_vx = initial_vx;
+            target_vy = initial_vy;
+            shade_workspace(x, y, true);
+            shade_workspace(target_vx, target_vy, false);
+            deactivate();
+            return;
+
+          case KEY_UP:
+            target_vy -= 1;
+            break;
+
+          case KEY_DOWN:
+            target_vy += 1;
+            break;
+
+          case KEY_RIGHT:
+            target_vx += 1;
+            break;
+
+          case KEY_LEFT:
+            target_vx -= 1;
+            break;
+
+          default:
+            return;
+        }
+
+        /* this part is only reached if one of the arrow keys is pressed */
+        if (key != key_pressed)
+        {
+            if (key_pressed)
+            {
+                timer_delay.disconnect();
+                timer_rate.disconnect();
+            }
+
+            timer_delay.set_timeout(delay, [=] ()
+            {
+                timer_rate.set_timeout(1000 / rate, [=] ()
+                {
+                    if (!should_handle_key())
+                    {
+                        // disconnect if key events should no longer be handled
+                        key_pressed = 0;
+                        return false;
+                    }
+
+                    handle_key_pressed(key);
+                    return true; // repeat
+                });
+
+                return false; // no more repeat
+            });
+
+            key_pressed = key;
+        }
+
+        auto dim = output->workspace->get_workspace_grid_size();
+        if (target_vx >= dim.width)
```

Comment:
What about using `(target_vx + dim.width) % dim.width` instead of this series of ifs?